lecture_2/05_offerfeed/internal/tasks/run.go

```diff
@@ -2,15 +2,35 @@ package tasks
 
 import (
 	"context"
+	"log"
+	"sync"
 )
 
 func RunTasks(tasks ...Task) {
-	// run each task in separate goroutine
+	wg := &sync.WaitGroup{}
+	wg.Add(len(tasks))
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for i, task := range tasks {
+		// running each task in separate goroutine
+		go func(i int, task Task) {
+			// when first task finishes, signals to the other goroutines that application should stop
+			defer wg.Done()
+			defer cancel()
+
+			_ = task.Start(ctx)
```

Comment:
Catch error here and handle it properly. (Probably just print it out as an information)